skrub/_interpolation_joiner.py

```diff
@@ -392,14 +411,17 @@ def _handles_multioutput(estimator):
 
 
 def _fit(key_values, target_table, estimator, propagate_exceptions):
+    target_table = std(target_table)
     estimator = clone(estimator)
-    kept_rows = target_table.notnull().all(axis=1).to_numpy()
+    ns = skrubns(target_table.dataframe)
+    kept_rows = ~(std(ns.any_rowwise(target_table.is_null().dataframe)).to_array())
     key_values = key_values[kept_rows]
-    Y = target_table.to_numpy()[kept_rows]
+    target_table = target_table.persist()
+    Y = target_table.to_array(None)[kept_rows]
```

Comment:
For readability
```suggestion
    Y = target_table.to_array(dtype=None)[kept_rows]
```

Reply:
I also need to change this in the standard, I think the output `dtype` should always be inferrable from the dtypes of the columns